### PR TITLE
feat: default multi-collateral token for transfer token

### PR DESCRIPTION
**Pull request by Xaroz**
This PR implements a way to allow a default multi collateral route to be set as the default

- Rename `getLowestFeeTransferToken` to `getTransferToken`
- Add `defaultMultiCollateralRoutes.ts` file where the default routes will be set
- Implement `tryGetDefaultOriginToken` which returns a multi collateral token based on the default MCWR object
- Update tests and README

fixes [ENG-2771](https://linear.app/hyperlane-xyz/issue/ENG-2771/have-nexus-default-exclusively-to-first-party-warp-route)

## Summary by CodeRabbit

* **New Features**
  * Added support for configurable default multi-collateral routes, allowing users to specify preferred collateral selections for transfers instead of relying solely on fee-based selection.

* **Documentation**
  * Added guidance on customizing default multi-collateral warp route behavior.

* **Tests**
  * Added comprehensive test coverage for default multi-collateral route selection logic and fallback behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<h3>Greptile Summary</h3>

This PR adds a configuration-driven way to set default multi-collateral warp routes, bypassing the fee-based selection when a preferred route is configured. The implementation renames `getLowestFeeTransferToken` to `getTransferToken` to better reflect its broader purpose.

## What Changed
- Introduced `defaultMultiCollateralRoutes.ts` config file pre-populated with first-party USDC routes across 7 chains
- Added `tryGetDefaultOriginToken()` function that checks the config before falling back to fee-based selection
- Updated `getTransferToken()` to prioritize configured defaults over lowest-fee logic
- Added comprehensive test coverage including edge cases for native tokens and fallback behavior
- Updated documentation in `CUSTOMIZE.md` to explain the override mechanism

## Implementation Quality
The logic properly handles both ERC20 collateral tokens and HypNative tokens (using 'native' as the key), with thorough null checks and early returns throughout the chain. The fee lookup is correctly bypassed when a default route is found, which improves performance for configured routes. Test coverage looks solid with proper mocking and validation of both the happy path and various edge cases.

<details><summary><h3>Confidence Score: 5/5</h3></summary>

- This PR is safe to merge with minimal risk
- The changes are well-structured with proper separation of concerns, comprehensive test coverage (157 lines of new tests), and backward compatibility (defaults are optional and fall back to existing fee-based logic). The function rename from getLowestFeeTransferToken to getTransferToken better reflects the new behavior, and all call sites were correctly updated.
- No files require special attention
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/consts/defaultMultiCollateralRoutes.ts | New config file that maps chains to their default multi-collateral routes for USDC across 7 chains |
| src/features/tokens/types.ts | Added DefaultMultiCollateralRoutes type definition to support route configuration mapping |
| src/features/tokens/utils.ts | Added tryGetDefaultOriginToken function with comprehensive logic for looking up configured default routes |
| src/features/transfer/fees.ts | Renamed getLowestFeeTransferToken to getTransferToken, added defaultMultiCollateralRoutes param with priority lookup |

</details>

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant User
    participant TransferForm as TransferTokenForm
    participant GetTransfer as getTransferToken
    participant TryDefault as tryGetDefaultOriginToken
    participant GetSameCollateral as getTokensWithSameCollateralAddresses
    participant WarpCore
    participant DefaultConfig as defaultMultiCollateralRoutes

    User->>TransferForm: Submit transfer form
    TransferForm->>GetTransfer: getTransferToken(originToken, destinationToken, defaultMultiCollateralRoutes)
    
    GetTransfer->>GetTransfer: isValidMultiCollateralToken?
    alt Not valid multi-collateral
        GetTransfer-->>TransferForm: Return originToken (no change)
    end
    
    GetTransfer->>GetSameCollateral: getTokensWithSameCollateralAddresses()
    GetSameCollateral-->>GetTransfer: List of tokens with same collateral
    
    alt Only 1 token found
        GetTransfer-->>TransferForm: Return originToken
    end
    
    GetTransfer->>TryDefault: tryGetDefaultOriginToken(originToken, destinationToken, config, tokens)
    TryDefault->>DefaultConfig: Lookup origin chain config
    TryDefault->>DefaultConfig: Lookup destination chain config
    TryDefault->>DefaultConfig: Get default warp route addresses
    
    alt Default route configured and found
        TryDefault-->>GetTransfer: Return default token
        GetTransfer-->>TransferForm: Return default token (bypass fee lookup)
    else No default configured
        TryDefault-->>GetTransfer: Return null
        GetTransfer->>WarpCore: getTokenCollateral() for each token
        WarpCore-->>GetTransfer: Balance results
        GetTransfer->>GetTransfer: Filter tokens with sufficient balance
        GetTransfer->>WarpCore: getInterchainTransferFee() for each token
        WarpCore-->>GetTransfer: Fee quotes
        GetTransfer->>GetTransfer: Sort by lowest fee
        GetTransfer-->>TransferForm: Return lowest fee token
    end
```
</details>

### CODE REVIEW

File: src/consts/defaultMultiCollateralRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// Default multi-collateral warp route configuration
+// Maps: chainName -> collateralAddress -> warpRouteAddressOrDenom
+//
+// For ERC20/collateralized tokens:
+//   { "ethereum": { "0xUSDC...": "0xWarpRoute..." } }
+//
+// For native tokens (HypNative), use 'native' as the key:
+//   { "ethereum": { "native": "0xWarpRoute..." } }
+export const defaultMultiCollateralRoutes: DefaultMultiCollateralRoutes | undefined = {
```
Comment (Xaroz) [inline src/consts/defaultMultiCollateralRoutes.ts]: will set to undefined when this is merged and will be added to nexus instead
